custom/ilsgateway/views.py

```diff
@@ -327,11 +327,10 @@ def get_success_url(self):
 class DashboardPageRedirect(RedirectView):
 
     @method_decorator(login_and_domain_required)
-    def dispatch(self, request, *args, **kwargs):
-        return super(DashboardPageRedirect, self).dispatch(request, *args, **kwargs)
+    def dispatch(self, request, domain, *args, **kwargs):
+        return super(DashboardPageRedirect, self).dispatch(request, domain, *args, **kwargs)
 
-    def get_redirect_url(self, *args, **kwargs):
-        domain = kwargs['domain']
+    def get_redirect_url(self, domain, *args, **kwargs):
```

Comment:
(the introduction of the `login_and_domain_required` decorator explicitly turns domain into an arg rather than a kwarg)